### PR TITLE
Indigo dev

### PR DESCRIPTION
**Pull request by ipa-jsf**
indigo_dev build.
Merged changes from hydro_dev.
Successfully build and catkin_lint it
@ipa-fxm Please Review

### CODE REVIEW

File: cob_perception_common/common/include/cob_perception_common/StdAfx.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#include <boost/random.hpp>
+#include <boost/mpl/size.hpp>
+
+#include <opencv/cv.h>
+#include <opencv/ml.h>
+#include <opencv/highgui.h>
+#include <opencv2/flann/hierarchical_clustering_index.h>
+
+#include "cob_object_perception_intern/windows/src/extern/TinyXml/tinyxml.h"
+#include "cob_object_perception_intern/windows/src/extern/TinyXml/tinystr.h"
```
Comment (fmessmer) [inline cob_perception_common/common/include/cob_perception_common/StdAfx.h]: Dependency to an internal package `cob_object_perception_intern`
This won't work...

Why the windows reference?
Can't you use the official (Linux) `TinyXml` system dependency?

File: cob_perception_common/common/src/StdAfx.cpp [hunk 2]
```diff
@@ -0,0 +1 @@
+#include "../include/cob_perception_common/StdAfx.h"
```
Comment (fmessmer) [inline cob_perception_common/common/src/StdAfx.cpp]: Where is this file needed?
What do you do with this cpp?

File: cob_vision_utils/common/src/CameraSensorToolbox.cpp [hunk 3]
```diff
@@ -50,17 +50,17 @@
 * If not, see <http://www.gnu.org/licenses/>.
 *
 ****************************************************************/
-//#include "../../../../cob_object_perception_intern/windows/src/PreCompiledHeaders/StdAfx.h"
+#include "../../../cob_perception_common/common/include/cob_perception_common/StdAfx.h"
```
Comment (fmessmer) [inline cob_vision_utils/common/src/CameraSensorToolbox.cpp]: Please no relative paths!!!
Use the <_package_name_/_header.h_> syntax!

**Comment by fmessmer** [comment]
The `cob_perception_common` package in this repo is a meta-package!
There should not be any code file in it!
`cob_vision_utils``is probably the better location for StdAfx.h and Co...

**Comment by fmessmer** [comment]
More review to come after the first issues have been adressed...
Should not be merged like this!

**Comment by ipa-jsf** [comment]
@ipa-fmw: Concerning StdAfx.h file and relative paths: StdAfx is a precompiled header file, that is used under Windows to speed up compilation. Under Linux it is ignored (see the preprocessor directives). The path to the header file must be the first line of code, therefore I cannot put it in between the windows preprocessor tags, and therefore the path is expressed relative.

I know, there could be problems when installing the package, because the path may no longer be valid. I will check under visual studio if I can properly set an additional include path in order to comply with the ROS syntax.

**Comment by ipa-jsf** [comment]
@ipa-fxm Fixed issue with precompiled header files

File: cob_vision_utils/common/include/cob_vision_utils/CameraSensorToolbox.h [hunk 4]
```diff
@@ -70,7 +68,7 @@
 	#include "cob_perception_common/cob_vision_utils/common/include/cob_vision_utils/CameraSensorTypes.h"
```
Comment (fmessmer) [inline cob_vision_utils/common/include/cob_vision_utils/CameraSensorToolbox.h:68]: This line is a duplicate of the one before!?

**Comment by fmessmer** [comment]
I'm still not sure, why the commits for this PR contain commits referring to merging with ELECTRIC?

File: cob_vision_utils/common/include/cob_vision_utils/CameraSensorToolbox.h (hunk 4 above)
Comment (ipa-jsf) [inline cob_vision_utils/common/include/cob_vision_utils/CameraSensorToolbox.h:68]: @ipa-fxm Yes, but only used for windows

**Comment by ipa-jsf** [comment]
@ipa-fxm Me neither, but I pulled from ipa320/groovy_dev and ipa320/indigo_dev before comitting. Should be alright.

File: cob_vision_utils/common/include/cob_vision_utils/CameraSensorToolbox.h (hunk 4 above)
Comment (fmessmer) [inline cob_vision_utils/common/include/cob_vision_utils/CameraSensorToolbox.h:68]: Ah, ok...in the diff view the #else is not shown....ok then...

**Comment by fmessmer** [comment]
LGTM then...

**Comment by floweisshardt** [comment]
visual review: Ok
